test/performance/shmem_perf_suite/bw_common.h

```diff
@@ -781,3 +811,58 @@ void static inline uni_dir_bw_main(int argc, char *argv[], bw_style bwstyl) {
     if (ret != -1)
         bw_finalize();
 } /*main() */
+
+static inline int check_hostname_validation(perf_metrics_t my_info) {
+
+    int hostname_status = -1;
+    int hostname_size = (HOST_NAME_MAX % 4 == 0) ? HOST_NAME_MAX : HOST_NAME_MAX + (4 - HOST_NAME_MAX % 4);
+    int i,j;
+
+    static long pSync_collect[SHMEM_COLLECT_SYNC_SIZE];
+    for (i = 0; i < SHMEM_COLLECT_SYNC_SIZE; i++)
+        pSync_collect[i] = SHMEM_SYNC_VALUE;
+
+    char *hostname = (char *) shmem_malloc (hostname_size * sizeof(char));
+    char *dest = (char *) shmem_malloc (my_info.num_pes * hostname_size * sizeof(char));
```

Comment:
Should these buffers be freed before this function returns?

Reply:
Actually, I had it. But then removed it because of a crash caused by multiple free of a single location. Thanks for catching this. Would check that back in.